internal/command/local/override/override_readiness.go

```diff
@@ -0,0 +1,199 @@
+				return
+			}
+			if !readiness.Ready() {
+				fmt.Fprintf(w, "sandbox not ready...\n")
+			} else if !ready {
+				ready = true
+				fmt.Fprintf(w, "sandbox ready.\n")
+
+			}
+			for err := readiness.Warn(); err != nil; err = readiness.Warn() {
+				fmt.Fprintf(w, "warning: %v\n", err)
+			}
+		}
+	}
+}
+
+func ckMatch(cfg *config.LocalOverrideCreate, sb *models.Sandbox, override string) func() (bool, error, error) {
+	return func() (bool, error, error) {
+		obs, err := getSandbox(cfg)
+		if err != nil {
+			return false, nil, err
+		}
+		ready := obs.Status.Ready
+		if err := ckAllMiddleware(sb.Spec.Middleware, obs.Spec.Middleware, sb.Name, override); err != nil {
+			return ready, nil, err
+		}
+		if err := ckForwards(sb.Spec.Routing, obs.Spec.Routing, sb.Name, override); err != nil {
+			return ready, nil, err
+		}
+
+		if !ready {
+			return false, fmt.Errorf("sandbox %s is not ready to accept requests", sb.Name), nil
+		}
+		return true, nil, nil
+	}
+}
+
+func ckAllMiddleware(desMWs, obsMWs []*models.SandboxesMiddleware, sbName, oName string) error {
+	desOMWs := getOverrideMWs(desMWs, oName)
+	obsOMWs := getOverrideMWs(obsMWs, oName)
+	if len(obsOMWs) < len(desOMWs) {
+		return fmt.Errorf("sandbox %s is missing override middleware", sbName)
+	}
+	if len(obsOMWs) > len(desOMWs) {
+		return fmt.Errorf("sandbox %s has interfering override middleware", sbName)
+	}
+	for i := range desOMWs {
+		obs, des := obsOMWs[i], desOMWs[i]
+		if err := ckMiddleware(des, obs, sbName, oName); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func ckMiddleware(des, obs *models.SandboxesMiddleware, sbName, oName string) error {
+	if len(obs.Args) < len(des.Args) {
+		return fmt.Errorf("sandbox %s missing override middleware arguments", sbName)
+	}
+	// args
+	argMap := map[string]*models.SandboxesArgument{}
+	for _, da := range des.Args {
+		argMap[da.Name] = da
+	}
+	for _, oa := range obs.Args {
+		da := argMap[oa.Name]
+		if da == nil {
+			return argConflict(sbName, oa.Name)
+		}
+		if da.Value != oa.Value {
+			return argConflict(sbName, oa.Name)
+		}
+		if (da.ValueFrom == nil) != (oa.ValueFrom == nil) {
+			return argConflict(sbName, oa.Name)
+		}
+		if da.ValueFrom == nil {
+			continue
+		}
+		if da.ValueFrom.Forward != oa.ValueFrom.Forward {
+			return argConflict(sbName, oa.Name)
+		}
+	}
+	// matches
+	if len(obs.Match) != len(des.Match) {
+		return fmt.Errorf("sandbox %s has unexpected match criteria", sbName)
+	}
+	mMap := map[string]bool{}
+	for _, match := range des.Match {
+		mMap[match.Workload] = true
+	}
+	for _, match := range obs.Match {
+		if mMap[match.Workload] {
+			continue
+		}
+		return fmt.Errorf("sandbox %s has unexpected match criteria", sbName)
+	}
+	return nil
+}
+
+func argConflict(sbName, argName string) error {
+	return fmt.Errorf("sandbox %s has conflicting middleware argument %q", sbName, argName)
+}
+
+func getOverrideMWs(mws []*models.SandboxesMiddleware, oName string) []*models.SandboxesMiddleware {
```

Comment:
Why this returns `[]*models.SandboxesMiddleware` instead of `*models.SandboxesMiddleware`?